core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java

```diff
@@ -0,0 +1,367 @@
+
+    /**
+     * Packages removed from java8 up to java11
+     * https://blog.codefx.org/java/java-11-migration-guide/
+     */
+    private final static String[] MOVED_PACKAGES = new String[] {"javax.activation", "javax.annotation", "javax.jws",
+            "javax.rmi", "javax.transaction", "javax.xml.bind", "javax.xml.soap", "javax.xml.ws", "org.omg",
+            "javax.activity", "com.sun", "sun"};
+
+    /**
+     * Places where a ClassNotFoundException is going to be thrown but it's ignored later in the code, so we
+     * don't have to send this exception, even though it might be related with java classes of moved packages
+     */
+    private static String[][] IGNORED_PLACES = {
+            {"hudson.util.XStream2$AssociatedConverterImpl", "findConverter"},
+            {"org.jenkinsci.plugins.workflow.cps.global.GrapeHack", "hack"},
+            {"org.codehaus.groovy.runtime.callsite.CallSiteArray", "createCallStaticSite"},
+            {"groovy.lang.MetaClassImpl", "addProperties"},
+            // We set the reportException call directly in this method when it's appropriated
+            {"hudson.PluginManager.UberClassLoader", "findClass"},
+            {"hudson.ExtensionFinder$GuiceFinder$FaultTolerantScope$1", "get"},
+            {"hudson.ExtensionFinder$GuiceFinder$SezpozModule", "resolve"},
+            {"java.beans.Introspector", "findCustomizerClass"},
+            {"com.sun.beans.finder.InstanceFinder", "instantiate"},
+            {"com.sun.beans.finder.ClassFinder", "findClass"},
+            {"java.util.ResourceBundle$Control", "newBundle"},
+            //hundreds when a job is created
+            {"org.codehaus.groovy.control.ClassNodeResolver", "tryAsLoaderClassOrScript"},
+            {"org.kohsuke.stapler.RequestImpl$TypePair", "convertJSON"}
+    };
+
+    @Nonnull
+    @Override
+    public String getDisplayName() {
+        return "Missing classes related with Java updates";
+    }
+
+    @Nonnull
+    @Override
+    public LocalDate getStart() {
+        return START;
+    }
+
+    @Nonnull
+    @Override
+    public LocalDate getEnd() {
+        return END;
+    }
+
+    /**
+     * To allow asserting this info in tests.
+     * @return the events gathered.
+     */
+    @VisibleForTesting
+    /* package */ static MissingClassEvents getEvents() {
+        return events;
+    }
+
+    /**
+     * This telemetry is only enabled when running on Java versions newer than Java 8.
+     * @return true if running on a newer Java version than Java 8
+     */
+    public static boolean enabled() {
+        return JavaUtils.getCurrentJavaRuntimeVersionNumber().isNewerThan(JavaSpecificationVersion.JAVA_8);
+    }
+
+    @CheckForNull
+    @Override
+    public JSONObject createContent() {
+        // If we are on the time window of this telemetry but we are not running on Java > 1.8, we don't send anything
```

Comment:
The time window is not checked later either.

Reply:
The comment says we don't send anything depending on the Java version and the time windows, but the enabled() method is checking only the Java version. If it's being checked anywhere else, then the comment needs to be changed in my opinion.

Reply:
ok, changed.